### PR TITLE
chore: 3d-tile & i3s loaders refactoring

### PR DESCRIPTION
**Pull request by atomicra**

### CODE REVIEW

File: modules/tiles/src/tileset/helpers/bounding-volume.js [hunk 1]
```diff
@@ -28,101 +41,30 @@ const scratchTransform = new Matrix4();
...
-      new Matrix4()
-    );
-  }
 
-  if (boundingVolumeHeader.sphere) {
-    return createSphere(boundingVolumeHeader.sphere, transform, result);
-  }
+  switch (true) {
+    case box != void 0:
+      return OBB_transform(box.clone(), transform);
```
Comment (ibgreen) [inline modules/tiles/src/tileset/helpers/bounding-volume.js:53]: Do we need to clone()? Are we keeping the old object?

File: modules/tiles/src/tileset/helpers/bounding-volume.js [hunk 2]
```diff
@@ -2,24 +2,37 @@
...
 
-// const scratchProjectedBoundingSphere = new BoundingSphere();
+const scratchVector = new Vector3();
+const transformScratchVectorU = new Vector3();
+const transformScratchVectorV = new Vector3();
+const transformScratchVectorW = new Vector3();
 
-function defined(x) {
-  return x !== undefined && x !== null;
+function BS_transform(obj, transform) {
```
Comment (ibgreen) [inline modules/tiles/src/tileset/helpers/bounding-volume.js:14]: We don't use underscores, or abbreviations (unless very common). We use `verbNoun` convention for functions. `transformBoundingVolume`?

File: modules/tiles/src/tileset/helpers/bounding-volume.js [hunk 3]
```diff
@@ -2,24 +2,37 @@
...
 }
 
-// const scratchMatrix = new Matrix3();
-const scratchScale = new Vector3();
-const scratchNorthWest = new Vector3();
-const scratchSouthEast = new Vector3();
-// const scratchRectangle = new Rectangle();
-// const scratchOrientedBoundingBox = new OrientedBoundingBox();
-const scratchTransform = new Matrix4();
+function OBB_transform(obj, transformation) {
```
Comment (ibgreen) [inline modules/tiles/src/tileset/helpers/bounding-volume.js:22]: `transformOBB`?

File: modules/i3s/src/lib/parsers/parse-i3s.js [hunk 4]
```diff
@@ -27,30 +29,45 @@ export function normalizeTileData(tile, options, context) {
...
+    : tile.mbs;
+  const cartesianCenter = Ellipsoid.WGS84.cartographicToCartesian([x, y, z]);
+  const cartesianCenterMbs =
+    tile.obb && tile.mbs
+      ? Ellipsoid.WGS84.cartographicToCartesian(tile.mbs.slice(0, 3))
+      : cartesianCenter;
+  const radiusMbs = tile.obb && tile.mbs ? tile.mbs[3] : radius;
+  tile.mbs = tile.mbs ? tile.mbs : [x, y, z, radius];
+
+  const box = tile.obb
```
Comment (ibgreen) [inline modules/i3s/src/lib/parsers/parse-i3s.js:60]: No need for ternary `:?`:

`const box = tile.obb && createBox...`

File: modules/tiles/src/tileset/helpers/bounding-volume.js (hunk 2 above)
Comment (ibgreen) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:14]: Please add JSDoc headers to all functions.

**Review by ibgreen — COMMENTED** [review]
I see no tests in the PR. Better check if there is any existing test coverage for this.

File: modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js [hunk 5]
```diff
@@ -42,6 +72,27 @@ export function normalizeTileData(tile, options) {
   tile.transformMatrix = tile.transform;
   tile.type = getTileType(tile);
   tile.refine = getRefine(tile.refine);
+
+  // =====================================
+  const _box = tile.boundingVolume && tile.boundingVolume.box;
```
[the remark this answers is not in the document]
Comment (atomicra) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js:77]: ok, how about boxArr, sphereArr?

File: modules/tiles/src/tileset/helpers/bounding-volume.js (hunk 1 above)
Comment (atomicra) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:53]: The one instance relate to many tiles. That is fixed the bug of a correct tiles location.

File: modules/tiles/src/tileset/helpers/bounding-volume.js (hunk 2 above)
Comment (atomicra) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:14]: ok, I'll fix it. This is temporary functions. The next steep will be PR for math.gl (it has done)

File: modules/tiles/src/tileset/helpers/bounding-volume.js (hunk 3 above)
Comment (atomicra) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:22]: ok.

File: modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js [hunk 6]
```diff
@@ -31,6 +39,28 @@ function getRefine(refine) {
   }
 }
 
+function createRegion(_region) {
```
[the remark this answers is not in the document]
Comment (atomicra) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js:42]: ok

File: modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js:77]: We don't use abbreviations. Please type out the words. I have no idea what `Arr` means :)

[the remark this answers is not in the document]
Comment (atomicra) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js:77]: tile.boundingVolume.box = Array(12)[ .. ]

[the remark this answers is not in the document]
Comment (atomicra) [reply modules/3d-tiles/src/lib/parsers/parse-3d-tile-header.js:77]: tile.boundingVolume.sphere = Array(4)[ .. ]

File: modules/tiles/src/tileset/helpers/bounding-volume.js (hunk 2 above)
Comment (atomicra) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:14]: BS_transform --> transformBoundingSphere
OBB_transform --> transformOrientedBoundingBox
ok?

Comment (ibgreen) [reply modules/tiles/src/tileset/helpers/bounding-volume.js:14]: Love it!

**Pull request closed without merging**